lib/rubycritic/core/analysed_module.rb

```diff
@@ -27,6 +28,15 @@ def rating
       @rating ||= Rating.from_cost(cost)
     end
 
+    def lines
+      return @lines if @lines
+      count = 0
+      File.open(path).each_line do |line|
+        count += 1 unless line.strip[0] == "#" || line.strip.empty?
```

Comment:
Let's move the guard

```Ruby
line.strip[0] == "#" || line.strip.empty?
```

into an appropriately named method.
How about

```Ruby
def ignore_line?(line)
  line.strip[0] == "#" || line.strip.empty?
end
```

?

You could directly use File's `readlines` and Enumerable's `reject`, then this whole method could be written more concise like this:

```Ruby
def lines
  @lines ||= File.open(path).readlines
                                 .reject { |line| ignore_line?(line) }
                                  .size
```

Then you can also drop the "begin" I suggested above ;)